src/test/java/fakes/UserControllerTest.java

```diff
@@ -24,26 +48,95 @@ void withValidInexistingUsername_returnsOK__NO_FAKE_DEMO() {
 
     @Test
     void withValidInexistingUsername_returnsOK__FAKE() {
-      // TODO
+      // check
       // 1. Test schneller machen
       // 2. UserController.create so beinflussen,
       //    dass einmal der "if"- und einmal der "else"-Fall durchlaufen wird
+      // => der else-Fall findet sich in der Methode withValidExistingUsername_returnsNOTOK__FAKE
+
+      FakeUserValidator fakeUserValidator = new FakeUserValidator();
+      // User wurde nicht erkannt
+      fakeUserValidator.setUserOk(false);
+      UserController ctrl = new UserController(fakeUserValidator, new MockDatabase());
+      User user = new User("kalua");
+
+      Message result = ctrl.create(user);
+      Assertions.assertEquals(result.status, Message.Status.OK);
+
+    }
+
+    @Test
+    void withValidExistingUsername_returnsNOTOK__FAKE() {
+      // check
+      FakeUserValidator fakeUserValidator = new FakeUserValidator();
+      // User wurde erkannt
+      fakeUserValidator.setUserOk(true);
+      UserController ctrl = new UserController(fakeUserValidator, new MockDatabase());
+      User user = new User("kalua");
+
+      Message result = ctrl.create(user);
+      Assertions.assertEquals(result.status, Message.Status.NOT_OK);
     }
 
+
     @Test
     void withValidInexistingUsername_returnsOK__MOCKITO() {
       // TODO
+      Database mockDatabase = mock(Database.class);
+      final UserValidator uv = new UserValidator();
+
+      List users = new ArrayList();
+      users.add("Peter");
+      doReturn(users).when(mockDatabase).getUsers();
+
+      UserController mockCtrl = mock(UserController.class);
+
+      User user = new User("kalua");
+
+      Message result = mockCtrl.create(user);
+      Assertions.assertEquals(result.status, Message.Status.NOT_OK);
     }
 
     @Test
-    void withValidInexitingUsername_addUserToDB__FAKE() {
+    void returnsFalseIfUsernameNotInDBYet() {
+      // check
+      Database mockDatabase = mock(Database.class);
+      final UserValidator uv = new UserValidator();
+
+      List users = new ArrayList();
+      users.add("Peter");
+      doReturn(users).when(mockDatabase).getUsers();
+
+      final boolean usernameExists = uv.doesUsernameExist("Peter");
+
+      Assertions.assertFalse(usernameExists);
+    }
+    @Test
+    void withValidInexistingUsername_addUserToDB__FAKE() {
       // TODO
       // Der Test soll prüfen, ob der Benutzer tatsächlich der DB hinzugefügt wurde.
       // Dazu soll ein Database Mock Objekt verwendet werden.
+      FakeUserValidator fakeUserValidator = new FakeUserValidator();
+      // User wurde erkannt
+      fakeUserValidator.setUserOk(true);
+      MockDatabase database = new MockDatabase();
+      UserController ctrl = new UserController(fakeUserValidator, database);
+      User user = new User("peter");
+
+      Message result = ctrl.create(user);
+      Assertions.assertEquals(result.status, Message.Status.NOT_OK);
```

Comment:
Dieses Assert (User nicht erstellt) widerspricht dem Assert auf Zeile 135 (User in DB hinzugefügt) oder nicht?
Gemäss Testname sollte hier ein Status.OK asserted werden nicht?